test/test_helpers.py

```diff
@@ -169,3 +169,28 @@ def test_flatten_schema_to_pyarrow_schema_type_not_defined():
 
     with pytest.raises(NotImplementedError):
         flatten_schema_to_pyarrow_schema(in_dict, ["created_at"])
+
+
+def test_flatten_array_fields():
+    in_dict = {
+        "int_array": [1, 2, 3],
+        "array_of_array": [1, 2, [3, 4]],
```

Comment:
What's gonna happen with `["a", "b", ["c", "d"]]`?

Reply:
I've included a new test case, the result will be `'["a", {"b": "value"}, ["c", "d"]]'`